payjoin-cli/src/app/v1.rs

```diff
@@ -348,7 +348,8 @@ impl App {
                     .require_network(network)
                     .map_err(|e| Error::Server(e.into()))?
                     .script_pubkey(),
-            )?
+            )
+            .map_err(|e| Error::Server(anyhow!(e.to_string()).into()))?
```

Comment:
`substitute_receiver_script` now returns an `OutputSubstitutionError`, not a Server error. Although I suppose we could keep the `log:warn!()` here instead of failing, is that what you mean?

Reply:
I see, my mistake during review.

The `Error::Server(e.into())` syntax could be used if OutputSubstitutionEror implemented `std::error::Error`, which it probably should since it's indeed an error and the payjoin-cli environment has `std` available.

```rs
impl std::error::Error for OutputSubstitutionError {
    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
        match &self.0 {
            InternalOutputSubstitutionError::OutputSubstitutionDisabled(_) => None,
            InternalOutputSubstitutionError::NotEnoughOutputs => None,
            InternalOutputSubstitutionError::InvalidDrainScript => None,
        }
    }
}
``